pkg/apis/core/validation/validation.go

```diff
@@ -1999,15 +1994,16 @@ func ValidatePersistentVolumeUpdate(newPv, oldPv *core.PersistentVolume, opts Pe
 	}
 
 	// PersistentVolumeSource should be immutable after creation.
-	if !apiequality.Semantic.DeepEqual(newPv.Spec.PersistentVolumeSource, oldPv.Spec.PersistentVolumeSource) {
-		pvcSourceDiff := cmp.Diff(oldPv.Spec.PersistentVolumeSource, newPv.Spec.PersistentVolumeSource)
-		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "persistentvolumesource"), fmt.Sprintf("spec.persistentvolumesource is immutable after creation\n%v", pvcSourceDiff)))
-	}
-	allErrs = append(allErrs, ValidateImmutableField(newPv.Spec.VolumeMode, oldPv.Spec.VolumeMode, field.NewPath("volumeMode"))...)
+	newErr := apimachineryvalidation.ValidateImmutableFieldWithErr(
+		newPv.Spec.PersistentVolumeSource,
+		oldPv.Spec.PersistentVolumeSource,
+		field.Forbidden(field.NewPath("spec", "persistentvolumesource"), "spec.persistentvolumesource is immutable after creation\n"))
```

Comment:
Also, you've included the `\n` here at the end, but not below. I think I would advise against it.